tests/test_mq_service.py

```diff
@@ -1362,3 +1363,25 @@ def test_owner_id_present_in_existing_host_but_missing_from_payload(mq_create_or
     assert updated_key == created_key
     assert updated_event["host"]["system_profile"]["owner_id"] == NEW_CN
     assert updated_event["host"]["display_name"] == "better_test_host"
+
+
+@pytest.mark.parametrize(
+    "user",
+    (
+        {"user": {"email": "tuser@redhat.com", "first_name": "test"}},
+        {"user": {"email": "tuser@redhat.com"}},
+        {"user": {"first_name": "test"}},
+        {"user": {}},
+        {},
+    ),
+)
+def test_create_host_by_user_with_missing_details(mq_create_or_update_host, db_get_host, user):
+    user_id = deepcopy(USER_IDENTITY)
+    user_id.update(user)
```

Comment:
Debugging this test revealed that if the "user" field isn't supplied, it keeps the one specified in `USER_IDENTITY`. Because of this, we need to delete that user field before we update it:
```suggestion
    user_id = deepcopy(USER_IDENTITY)
    del user_id["user"]
    user_id.update(user)
```